terraform/aptos-node/gcp/README.md

```diff
@@ -0,0 +1,180 @@
+# Using Terraform
+
+## Run on GCP
+This guide assumes you already have GCP account setup, and have created a new project for deploying Aptos node.
+
+Install pre-requisites if needed:
+
+   * Terraform 1.1.7: https://www.terraform.io/downloads.html
+   * Kubernetes cli: https://kubernetes.io/docs/tasks/tools/
+   * Google Cloud cli: https://cloud.google.com/sdk/docs/install-sdk
+
+1. Create a working directory for your configuration.
+
+    * Choose a workspace name e.g. `testnet`. Note: this defines terraform workspace name, which in turn is used to form resource names.
+    ```
+    $ export WORKSPACE=testnet
+    ```
+
+    * Create a directory for the workspace
+    ```
+    $ mkdir -p ~/$WORKSPACE
+    ```
+2. Create a storage bucket for storing the Terraform state on Google Cloud Storage.  Use the console or this gcs command to create the bucket.  The name of the bucket must be unique.  See the Google Cloud Storage documentation here: https://cloud.google.com/storage/docs/creating-buckets#prereq-cli
+  ```
+  $ gsutil mb gs://BUCKET_NAME
+  # for example
+  $ gsutil mb gs://<project-name>-aptos-terraform-dev
+  ```
+
+3. Create Terraform file called `main.tf` in your working directory:
+  ```
+  $ cd ~/$WORKSPACE
+  $ touch main.tf
+  ```
+
+4. Modify `main.tf` file to configure Terraform, and create fullnode from Terraform module. Example content for `main.tf`:
+  ```
+  terraform {
+    required_version = "~> 1.1.0"
+    backend "gcs" {
+      bucket = "BUCKET_NAME" # bucket name created in step 2
+      prefix = "state/aptos-node"
+    }
+  }
+
+  module "aptos-node" {
+    # download Terraform module from aptos-labs/aptos-core repo
+    source        = "github.com/aptos-labs/aptos-core.git//terraform/aptos-node/gcp?ref=testnet"
+    region        = "us-central1"  # Specify the region
+    zone          = "c"            # Specify the zone suffix
+    project       = "<GCP Project Name>" # Specify your GCP project name
+    era           = 1              # bump era number to wipe the chain
+    image_tag     = "testnet" # Specify the docker image tag to use
+    validator_name = "<Name of Your Validator, no space>"
+  }
+  ```
+
+5. Initialise Terraform in the same directory of your `main.tf` file
+  ```
+  $ terraform init
+  ```
+This should download all the terraform dependencies for you, in the `.terraform` folder.
+
+6. Create a new Terraform workspace to isolate your environments:
+  ```
+  $ terraform workspace new $WORKSPACE
+  # This command will list all workspaces
+  $ terraform workspace list
+  ```
+
+7. Apply the configuration.
+
+  ```
+  $ terraform apply
+  ```
+
+  This might take a while to finish (10 - 20 minutes), Terraform will create all the resources on your cloud account. 
+
+8. Once terraform apply finishes, you can check if those resources are created:
+
+    - `kubectl get pods` this should have haproxy, validator and fullnode. with validator and fullnode pod `pending` (require further action in later steps)
```

Comment:
Will the kuberntes provider change your kube context for you? Maybe include a step here for that if not